plugin/trino-exasol/src/test/java/io/trino/plugin/exasol/TestExasolTypeMapping.java

```diff
@@ -279,9 +282,112 @@ private void testDate(ZoneId sessionZone)
+                "HASHTYPE(1025 BYTE)",
+                "the given size of HASHTYPE is too large. At most 1024 bytes are allowed");
+
+        // Too few bits (< 8)
+        testUnsupportedHashTypeDefinition(
+                "HASHTYPE(7 BIT)",
+                "the given size of HASHTYPE is too small. A minimum of 8 bits are required");
+
+        // Too many bits (> 8192)
+        testUnsupportedHashTypeDefinition(
+                "HASHTYPE(8193 BIT)",
+                "the given size of HASHTYPE is too large. At most 8192 bits are allowed");
+
+        // Bits not divisible by 8
+        testUnsupportedHashTypeDefinition(
+                "HASHTYPE(9 BIT)",
+                "Bit size of HASHTYPE has to be a multiple of 8");
+
+        // Invalid hex character
+        testUnsupportedHashTypeValue(
+                "HASHTYPE(4 BYTE)",
+                "'GGGGGGGG'",
+                "data exception - Invalid hash format");
+
+        // Too short for declared size (expecting 4 bytes = 8 hex chars, got 6)
+        testUnsupportedHashTypeValue(
+                "HASHTYPE(4 BYTE)",
+                "'AABBCC'",
+                "data exception - Invalid hash format");
+
+        // Too short for declared size (expecting 16 bytes = 32 hex chars, got 31)
+        testUnsupportedHashTypeValue(
+                "HASHTYPE(16 BYTE)",
+                "'550e8400-e29b-11d4-a716-44665544000'",
+                "data exception - Invalid hash format");
+
+        // Too long for declared size (expecting 4 bytes = 8 hex chars, got 10)
+        testUnsupportedHashTypeValue(
+                "HASHTYPE(4 BYTE)",
+                "'AABBCCDDEE'",
+                "data exception - Invalid hash format");
+
+        // Unexpected symbol inside
+        testUnsupportedHashTypeValue(
+                "HASHTYPE(4 BYTE)",
+                "'AABB-CCZZ'",
+                "data exception - Invalid hash format");
+    }
+
+    private void testUnsupportedHashTypeDefinition(
+            String exasolType,
+            String expectedException)
+    {
+        String tableName = "test_unsupported_hashtype_" + randomNameSuffix();
+        assertExasolSqlQueryFails(
+                "CREATE TABLE %s.%s (col %s)".formatted(TEST_SCHEMA, tableName, exasolType),
+                expectedException);
+    }
+
+    private void testUnsupportedHashTypeValue(
+            String exasolType,
+            String inputLiteral,
+            String expectedException)
+    {
+        String tableName = "test_unsupported_hashtype_" + randomNameSuffix();
+        try {
+            exasolServer.execute("CREATE TABLE %s.%s (col %s)".formatted(TEST_SCHEMA, tableName, exasolType));
+            assertExasolSqlQueryFails(
+                    "INSERT INTO %s.%s VALUES (%s)".formatted(TEST_SCHEMA, tableName, inputLiteral),
+                    expectedException);
+        }
+        finally {
+            exasolServer.execute("DROP TABLE %s.%s".formatted(TEST_SCHEMA, tableName));
```

Comment:
This statement may throw an exception if `CREATE TABLE` fails for some reason. Please add `IF EXISTS` option.